state/cloudcredentials.go

```diff
@@ -132,12 +133,16 @@ func (st *State) UpdateCloudCredential(tag names.CloudCredentialTag, credential
 		if err == nil {
 			ops = append(ops, updateCloudCredentialOp(tag, credential))
 		} else {
+			annotationMsg = "creating cloud credential"
+			if credential.Invalid || credential.InvalidReason != "" {
+				return nil, errors.NotValidf("adding invalid credential")
```

Comment:
Maybe NotSupported instead?
The credential itself is not Invalid from an error perspective, but adding such a credential is a not supported operation.